src/custom/hooks/useUSDCPrice/index.ts

```diff
@@ -0,0 +1,88 @@
+import { CurrencyAmount, Currency, Price, Token } from '@uniswap/sdk-core'
+import { useEffect, useMemo, useState } from 'react'
+import { unstable_batchedUpdates as batchedUpdate } from 'react-dom'
+import { supportedChainId } from 'utils/supportedChainId'
+import { useActiveWeb3React } from 'hooks/web3'
+import { getBestPrice } from 'utils/price'
+import { STABLECOIN_AMOUNT_OUT } from 'hooks/useUSDCPrice'
+import { stringToCurrency } from 'state/swap/extension'
+
+export * from '@src/hooks/useUSDCPrice'
+export { default } from '@src/hooks/useUSDCPrice'
+
+export function useBestUSDCPrice(currency?: Currency) {
+  const [bestUsdPrice, setBestUsdPrice] = useState<Price<Token, Currency> | null>(null)
+  const [error, setError] = useState<Error | null>(null)
+
+  const { chainId } = useActiveWeb3React()
+
+  // USDC constants
+  // 100_000e6 USDC @ 6 decimals
+  const amountOut = chainId ? STABLECOIN_AMOUNT_OUT[chainId] : undefined
+  const stablecoin = amountOut?.currency
+
+  useEffect(() => {
+    const isSupportedChain = supportedChainId(chainId)
+    if (!isSupportedChain || !currency || !amountOut || !stablecoin) return
+
+    const params = {
+      baseToken: stablecoin.address,
+      quoteToken: currency.wrapped.address,
+      kind: 'buy',
```

Comment:
not very important, but I just made a PR touching these to use the enum https://github.com/gnosis/cowswap/pull/902